src/core/cgroup.c

```diff
@@ -1847,7 +1847,8 @@ CGroupMask unit_get_target_mask(Unit *u) {
 
         own_mask = unit_get_own_mask(u);
 
-        if (own_mask & CGROUP_MASK_BPF_FIREWALL & ~u->manager->cgroup_supported)
+        if (!MANAGER_IS_TEST_RUN(u->manager) &&
+            own_mask & CGROUP_MASK_BPF_FIREWALL & ~u->manager->cgroup_supported)
```

Comment:
this series of && and & should really have some () to make clear what is what, even if C doesn't strictly require it.